### DISCUSSION TITLE
Формирование селекта по другой таблице с фильтрацией по селекту.

### DISCUSSION THREAD

**Discussion by LostAlly** [totumonline/totum-ru-issues-and-discussions#594] [Как это сделать?]
Здравствуйте.
Несколько дней бъюсь с задачей не могу победить уже кучу вариантов перепробовал, пока не дошел просто до проверки возвращаемых массивов )
Дано.
Таблица номеклатура пусть будет одно поле - имя (и плюс id)
Таблица спецификации - поле название спецификации(текстовое), поле номенклатура(селект из таблицы номенклатура)
Таблица заявка с полем данные формируемое через временную таблицу "состав заказа"и изменяемое по linkToTable.
В этой таблице выбирается номеклатура(селект из таблицы номеклатура), и спецификация отфильтрованная по полю номенклатура. 
Перепробовал много вариантов ничего не выходит....
Решил проверить что же возвращает selectRowListForSelect и увидел вот это(отформатировал)
```
[{"value":9,"is_del":false,"title":"PP\/860"},
{"value":10,"is_del":false,"title":"663"},
.....
{"value":4,"is_del":false,"title":"PP\/668"},
{"value":6,"is_del":false,"title":"PP\/668"},
{"value":7,"is_del":false,"title":"PP\/956"}]
```

Т.е. как-бы соотносит id шники верно, но почему не срабатывает where не пойму, по идее простейший запрос... 
Прям очень тяжко мне дается тотум ))) правда и времени я уделяю урывками...

**Comment by ttm-alexeypolunin** [comment]
Суть вопроса не понятна. В задаче есть:

**Номенклатура:**
```
- id - | - Название -
```

**Спецификация**
```
- Название - | - Номенклатура (сел) -
```

**Заявка:**
```
- Номенклатура (сел 1) - | - Спецификация (сел 2) -
```

Вам нужно, что бы в Заявке во втором селекте в Спецификация выбирались только те позиции, которые доступны при выбраном первом селекте Номенклатура?

У вас в этом месте where не срабатывает?

**Reply by ttm-alexeypolunin** [reply]
Если да, то ответ **Индивидуальный расчет Селекта для каждой строки**: https://ru.docs.totum.online/different-composition-of-select-depending-on-other-fields

**Reply by LostAlly** [reply]
Вы все верно поняли(Заявке во втором селекте в Спецификация выбирались только те позиции, которые доступны при выбраном первом селекте Номенклатура)

Забыл уточнить, вдруг это важно, селект номенклатуры это селект Дерева.

По логике должен работать такой запрос для формирования списка 
= : selectRowListForSelect(table: 'спецификации'; field: 'название_спецификации'; where: 'номеклатура_ спецификации' = #номеклатура_выбранная_в_пером стобце)
Галочка - [индивидуальный расчет селекта для каждой строки](https://ru.docs.totum.online/fields#fields-settings-codeSelectIndividual) стоит.

**Reply by LostAlly** [reply]
Разобрался - в столбце номенклатура убрал галочку - таблица строится по одной таблице, и заработал селект... очень странное поведение, как эта галочка влияет..
